src/engine.ts

```diff
@@ -32,39 +21,63 @@ export class JsonTemplateEngine {
   }
 
   private static compileAsAsync(
-    templateOrExpr: string | Expression,
+    templateOrExpr: string | Expression | FlatMappingPaths[],
     options?: EngineOptions,
   ): Function {
-    return CommonUtils.CreateAsyncFunction(
+    return CreateAsyncFunction(
       DATA_PARAM_KEY,
       BINDINGS_PARAM_KEY,
       this.translate(templateOrExpr, options),
     );
   }
 
+  private static translateExpression(expr: Expression): string {
+    const translator = new JsonTemplateTranslator(expr);
+    return translator.translate();
+  }
+
+  static parseMappingPaths(mappings: FlatMappingPaths[], options?: EngineOptions): Expression {
+    const flatMappingAST = mappings.map((mapping) => ({
+      input: JsonTemplateEngine.parse(mapping.input, options).statements[0],
+      output: JsonTemplateEngine.parse(mapping.output, options).statements[0],
+    }));
+    return convertToObjectMapping(flatMappingAST);
+  }
+
+  static create(
+    templateOrExpr: string | Expression | FlatMappingPaths[],
+    options?: EngineOptions,
+  ): JsonTemplateEngine {
+    return new JsonTemplateEngine(this.compileAsAsync(templateOrExpr, options));
+  }
+
+  static createAsSync(
+    templateOrExpr: string | Expression,
+    options?: EngineOptions,
+  ): JsonTemplateEngine {
+    return new JsonTemplateEngine(this.compileAsSync(templateOrExpr, options));
+  }
+
   static parse(template: string, options?: EngineOptions): Expression {
     const lexer = new JsonTemplateLexer(template);
     const parser = new JsonTemplateParser(lexer, options);
     return parser.parse();
   }
 
-  static translate(templateOrExpr: string | Expression, options?: EngineOptions): string {
-    if (typeof templateOrExpr === 'string') {
-      return this.translateTemplate(templateOrExpr, options);
+  static translate(
+    template: string | Expression | FlatMappingPaths[],
+    options?: EngineOptions,
+  ): string {
+    let templateExpr = template as Expression;
+    if (typeof template === 'string') {
+      templateExpr = this.parse(template, options);
+    } else if (Array.isArray(template)) {
```

Comment:
Instead of checking for array datatype, can explicit comparison of `FlatMappingPaths` object be used here ?